src/sentry/api/endpoints/organization_events.py

```diff
@@ -143,7 +144,25 @@ def data_fn(offset, limit):
                     "error": six.text_type(error),
                 },
             )
-            raise ParseError(detail="Invalid query.")
+            message = "Internal error. Please try again."
+            if type(error) in [
+                snuba.RateLimitExceeded,
+                snuba.QueryMemoryLimitExceeded,
+                snuba.QueryTooManySimultaneous,
+            ]:
+                message = "Query timeout. Please try again. If the problem persists try a smaller date range or fewer projects."
+            elif type(error) in [
+                snuba.UnqualifiedQueryError,
+                snuba.QueryExecutionError,
+                snuba.SchemaValidationError,
+            ]:
+                message = "Invalid query."
+            elif isinstance(error, snuba.QueryIllegalTypeOfArgument):
+                message = "Invalid query. Argument to function is wrong type."
```

Comment:
What can a user do with this information? Is there something they can do to help resolve it?

Reply:
Have better arguments? I debated whether to add this one, but I thought more specific would be better than less specific, even if the message is a little confusing.

Reply:
Ok. Lets run with this and see how it goes.